File: solutions/_2108_find_first_palindromic_string_in_the_array/solution.py
from typing import List


class Solution:
    def firstPalindrome(self, words: List[str]) -> str:
        def is_palindrome(word):
            return word == word[::-1]

        for word in words:
            if is_palindrome(word):
                return word
        return ""


# https://leetcode.com/problems/find-first-palindromic-string-in-the-array/

File: solutions/_2108_find_first_palindromic_string_in_the_array/solution.js
/**
 * @param {string[]} words
 * @return {string}
 */
var firstPalindrome = function (words) {
  const isPalindrome = (word) => {
    let left = 0;
    let right = word.length - 1;
    while (left < right) {
      if (word[left] !== word[right]) {
        return false;
      }
      left++;
      right--;
    }
    return true;
  };

  for (const word of words) {
    if (isPalindrome(word)) {
      return word;
    }
  }
  return "";
};

module.exports = { firstPalindrome };

/*
https://leetcode.com/problems/find-first-palindromic-string-in-the-array/
*/
